### PR TITLE
Refactor test observer naming for consistency

### PR DESCRIPTION
**Pull request by dusi**
# 📲 What

Refactor test observer naming

# 🤔 Why

We should name test observers consistently based on VM's properties

# 🛠 How

Mostly just removes `...Observer` suffix, but there was already 1 minor change to a VM who was using this suffix for no real reason.
I've made sure that test observer instances reflect their counterparts in VMs (so the test observers should now have the same name as VM signal)

# ✅ Acceptance criteria

- [x] All tests pass

### CODE REVIEW

File: Kickstarter-iOS/Views/Controllers/VideoViewController.swift [hunk 1]
```diff
@@ -88,10 +88,10 @@ public final class VideoViewController: UIViewController {
         }, completion: nil)
     }
 
-    self.viewModel.outputs.addCompletionObserver
```
Comment (dusi) [inline Kickstarter-iOS/Views/Controllers/VideoViewController.swift]: Remove `..Observer` suffix from the VM to stay consistent with the rest of the codebase

File: Kickstarter-iOS/Views/Controllers/VideoViewController.swift [hunk 2]
```diff
@@ -152,7 +152,7 @@ public final class VideoViewController: UIViewController {
     }
   }
 
-  func addCompletionObserver(atTime time: CMTime) {
+  func addCompletion(atTime time: CMTime) {
```
Comment (justinswart) [inline Kickstarter-iOS/Views/Controllers/VideoViewController.swift]: I would say that this should still have `Observer` in the name, the function is adding the completion observer.

File: Library/ViewModels/ChangePasswordViewModelTests.swift [hunk 3]
```diff
@@ -10,38 +10,38 @@ import XCTest
...
+  private let onePasswordButtonIsHidden = TestObserver<Bool, NoError>()
+  private let onePasswordFindPasswordForURLString = TestObserver<String, NoError>()
+  private let saveButtonIsEnabled = TestObserver<Bool, NoError>()
+  private let validationErrorLabelIsHidden = TestObserver<Bool, NoError>()
+  private let validationErrorLabelMessage = TestObserver<String, NoError>()
 
   override func setUp() {
     super.setUp()
 
-    self.vm.outputs.activityIndicatorShouldShow.observe(activityIndicatorShouldShowObserver.observer)
```
Comment (justinswart) [inline Library/ViewModels/ChangePasswordViewModelTests.swift]: I know it didn't have before but maybe you could add `self` to all of these?

File: Library/ViewModels/FindFriendsCellViewModelTests.swift [hunk 4]
```diff
@@ -12,32 +12,32 @@ import Foundation
...
+  private let disabledDescriptionLabelIsHidden = TestObserver<Bool, NoError>()
+  private let isDisabled = TestObserver<Bool, NoError>()
 
   override func setUp() {
     super.setUp()
 
     self.vm.outputs.disabledDescriptionLabelIsHidden
-      .observe(disabledDescriptionLabelIsHiddenObserver.observer)
-    self.vm.outputs.isDisabled.observe(isDisabledObserver.observer)
+      .observe(disabledDescriptionLabelIsHidden.observer)
```
Comment (justinswart) [inline Library/ViewModels/FindFriendsCellViewModelTests.swift]: `self`'s here too. SwiftFormat 😅

File: Library/ViewModels/FindFriendsFaceookConnectCellViewModelTests.swift [hunk 5]
```diff
@@ -32,7 +32,7 @@ import Prelude
     vm.outputs.attemptFacebookLogin.observe(attemptFacebookLogin.observer)
     vm.outputs.facebookConnectCellTitle.observe(title.observer)
     vm.outputs.facebookConnectCellSubtitle.observe(subtitle.observer)
-    vm.outputs.hideCloseButton.observe(hideCloseButtonObserver.observer)
+    vm.outputs.hideCloseButton.observe(hideCloseButton.observer)
```
Comment (justinswart) [inline Library/ViewModels/FindFriendsFaceookConnectCellViewModelTests.swift]: Here too but now I'm wondering if it should be a separate PR.

File: Kickstarter-iOS/Views/Controllers/VideoViewController.swift (hunk 1 above)
Comment (justinswart) [inline Kickstarter-iOS/Views/Controllers/VideoViewController.swift]: Yeah this should still have `Observer` in the name too.
